tests/unit/components/NcAvatar/NcAvatar.spec.ts

```diff
@@ -115,4 +115,90 @@ describe('NcAvatar.vue', () => {
 			expect(wrapper.text()).toBe(initials)
 		})
 	})
+
+	describe('Icon and image rendering', () => {
+		const avatarUrl = 'https://cloud.ltd/index.php/avatar/user1/64'
+
+		it('should render image with avatar url pointing to a user', async () => {
+			const wrapper = mount(NcAvatar, {
+				props: {
+					displayName: 'Guest',
+					user: 'user1',
+				},
+			})
+			/* Mock img.onload callback */
+			wrapper.vm.isAvatarLoaded = true
+			wrapper.vm.avatarUrlLoaded = avatarUrl
```

Comment:
Test should never manually change the internal state of the component.

Reply:
How to proceed then? Maybe mock an Image() constructor for these tests to emit load event?